src/components/Text/src/Text.vue

```diff
@@ -123,21 +132,35 @@ export default {
 		},
 		inlineStyles() {
 			const { fonts } = this.theme;
+
 			return {
-				fontFamily: this.resolvedFontFamily,
-				fontWeight: this.resolvedFontWeight,
+				fontFamily: this.useThemeComponentData ? this.resolvedFontFamily : this.fontFamily,
+				fontWeight: this.useThemeComponentData ? this.resolvedFontWeight : this.fontWeight,
 				color: this.resolvedColor,
```

Comment:
This no longer applies the correct resolved color. We would need to replace it with a computed property to handle mapping from `theme.colors.heading` and `theme.colors.text` to the different relevant variants.

OR the preferable approach would be to just change this to `color: this.color` and adding `--color-heading` and `--color-text` to the correct places in the CSS classes below.

I do think if we do this, we might as well remove `text` as a theme property - it wouldn't do anything with a useful level of specificity that Text.vue would't be getting from `theme.colors` and `theme.fonts` instead.

This is a pretty important distinction between this and the alternate approach which builds on our current theming system.

Reply:
@laurenhu You're right, I was working on some code that did that, but I was hesitant to change too much when changing how color applies didn't seem like the primary goal. But I think as-is it may be a regression or not work as expected. I'll do some more testing in the lab perhaps try both.

Reply:
I updated to leverage the color data in the style so the output maps as expected. It's also still possible to pass a color directly inline:
![CleanShot 2022-04-14 at 15 08 45](https://user-images.githubusercontent.com/129122/163485193-82acfd07-4718-4f69-b6b4-eb5148a77f61.png)